test/test_relational.py

```diff
@@ -365,6 +370,185 @@ def test_filter_spark_no_col_error(self):
+        trainable = GroupBy(by=[it["Product number"], it["Retailer code"]])
+        grouped_df = trainable.transform(self.go_sales[1]["go_daily_sales"])
+        self.assertEqual(grouped_df.ngroups, 5000)
+        trainable = Aggregate(
+            columns={
+                "mean_quantity": mean(it["Quantity"]),
+                "max_usp": max(it["Unit sale price"]),
+                "count_quantity": count(it["Quantity"]),
+            }
+        )
+        aggregated_df = trainable.transform(grouped_df)
+        self.assertEqual(aggregated_df.shape, (5000, 3))
+        self.assertEqual(
+            round(aggregated_df.loc[(130130, 1137), "mean_quantity"], 2), 8.74
+        )
+        self.assertEqual(aggregated_df.loc[(147160, 1192), "max_usp"], 32.85)
+        self.assertEqual(aggregated_df.loc[(147120, 1216), "count_quantity"], 52)
+
+    def test_aggregate_3(self):
+        trainable = GroupBy(by=[it["Product line"], it["Product brand"]])
+        grouped_df = trainable.transform(self.go_sales[3]["go_products"])
+        self.assertEqual(grouped_df.ngroups, 30)
+        trainable = Aggregate(
+            columns={
+                "sum_uc": sum(it["Unit cost"]),
+                "max_up": max(it["Unit price"]),
+                "max_uc": max(it["Unit cost"]),
+                "min_uc": min(it["Unit cost"]),
+                "min_up": min(it["Unit price"]),
+            }
+        )
+        aggregated_df = trainable.transform(grouped_df)
+        self.assertEqual(aggregated_df.shape, (30, 5))
+        self.assertEqual(
+            aggregated_df.loc[("Golf Equipment", "Blue Steel"), "max_uc"], 89.41
+        )
+        self.assertEqual(
+            aggregated_df.loc[("Outdoor Protection", "Sun"), "min_uc"], 1.79
+        )
+        self.assertEqual(
+            aggregated_df.loc[("Personal Accessories", "Edge"), "sum_uc"], 89.22
+        )
+        self.assertEqual(
+            aggregated_df.loc[("Mountaineering Equipment", "Granite"), "max_up"], 80
+        )
+        self.assertEqual(
+            aggregated_df.loc[("Personal Accessories", "Xray"), "min_up"], 125.4
+        )
+
+    def test_aggregate_no_col_error(self):
+        trainable = GroupBy(by=[it["Product number"]])
+        grouped_df = trainable.transform(self.go_sales[1]["go_daily_sales"])
+        self.assertEqual(grouped_df.ngroups, 244)
+        trainable = Aggregate(columns={"mean_quantity": mean(it["Quantity_1"])})
+        with self.assertRaises(KeyError):
+            _ = trainable.transform(grouped_df)
+
+
+# Testing aggregate operator for spark dataframes
+class TestAggregateSpark(unittest.TestCase):
+    # Get go_sales dataset in spark dataframe
+    def setUp(self):
+        self.go_sales_spark = fetch_go_sales_dataset("spark")
+        # Typecast numerical columns in the dataset which are of string datatype to float datatype
+        self.go_sales_spark[3]["go_products"] = (
+            self.go_sales_spark[3]["go_products"]
+            .withColumn(
```

Comment:
If `float` is the right type, wondering whether we should do this conversion in `fetch_go_sales_dataset`?

Reply:
As discussed, will include these changes in next PR.